Source: https://gitlab.com/ubports/development/core/click/-/commit/8f654978a12e6f9a0b6ff64296ec5565e3ff5cd0.patch
--
From 8f654978a12e6f9a0b6ff64296ec5565e3ff5cd0 Mon Sep 17 00:00:00 2001
From: Ratchanan Srirattanamet <ratchanan@ubports.com>
Date: Thu, 4 Apr 2024 01:46:32 +0700
Subject: [PATCH] preload: fix -Wimplicit-function-declaration

Copy the declarations of __fxstat{,64} from GLibC so that they're
properly declared.
---
 preload/clickpreload.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/preload/clickpreload.c b/preload/clickpreload.c
index 2799bff..ba6b9e7 100644
--- a/preload/clickpreload.c
+++ b/preload/clickpreload.c
@@ -439,6 +439,8 @@ int __xmknod (int ver, const char *pathname, mode_t mode, dev_t *dev)
     return (*libc___xmknod) (ver, pathname, mode, dev);
 }
 
+extern int __fxstat (int __ver, int __fildes, struct stat *__stat_buf);
+
 int __xstat (int ver, const char *pathname, struct stat *buf)
 {
     if (!libc___xstat)
@@ -450,6 +452,8 @@ int __xstat (int ver, const char *pathname, struct stat *buf)
     return (*libc___xstat) (ver, pathname, buf);
 }
 
+extern int __fxstat64 (int ver, int __fildes, struct stat64 *__stat_buf);
+
 int __xstat64 (int ver, const char *pathname, struct stat64 *buf)
 {
     if (!libc___xstat64)
-- 
GitLab

